lib/ruby_indexer/lib/ruby_indexer/rbs_indexer.rb

```diff
@@ -122,7 +127,102 @@ def handle_method(member, owner)
+      parameters = []
+      parameters += process_required_positionals(function) if function.required_positionals
+      parameters += process_optional_positionals(function) if function.optional_positionals
+      parameters += process_trailing_positionals(function) if function.trailing_positionals
+      parameters += process_rest_positionals(function) if function.rest_positionals
+      parameters += process_required_keywords(function) if function.required_keywords
+      parameters += process_optional_keywords(function) if function.optional_keywords
+      parameters += process_rest_keywords(function) if function.rest_keywords
+      parameters
+    end
+
+    sig { params(function: RBS::Types::Function).returns(T::Array[Entry::RequiredParameter]) }
+    def process_required_positionals(function)
+      function.required_positionals.map do |param|
+        # Some parameters don't have names, e.g.
+        #   def self.try_convert: [U] (untyped) -> ::Array[U]?
+        name = param.name || :object
+
+        Entry::RequiredParameter.new(name: name)
+      end
+    end
+
+    sig { params(function: RBS::Types::Function).returns(T::Array[Entry::OptionalParameter]) }
+    def process_optional_positionals(function)
+      function.optional_positionals.map.with_index do |param, i|
+        # Optional positionals may be unnamed, e.g.
+        #  def self.polar: (Numeric, ?Numeric) -> Complex
+        name = param.name || :"arg#{i}"
+
+        Entry::OptionalParameter.new(name: name)
+      end
+    end
+
+    sig { params(function: RBS::Types::Function).returns(T::Array[Entry::OptionalParameter]) }
+    def process_trailing_positionals(function)
+      function.trailing_positionals.map do |param|
+        Entry::OptionalParameter.new(name: param.name)
+      end
+    end
+
+    sig { params(function: RBS::Types::Function).returns(T::Array[Entry::RestParameter]) }
+    def process_rest_positionals(function)
+      rest = function.rest_positionals
+
+      rest_name = rest.name || Entry::RestParameter::DEFAULT_NAME
+
+      [Entry::RestParameter.new(name: rest_name)]
+    end
+
+    sig { params(function: RBS::Types::Function).returns(T::Array[Entry::KeywordParameter]) }
+    def process_required_keywords(function)
+      function.required_keywords.map do |name, _param|
+        Entry::KeywordParameter.new(name: name)
+      end
+    end
+
+    sig { params(function: RBS::Types::Function).returns(T::Array[Entry::OptionalKeywordParameter]) }
+    def process_optional_keywords(function)
+      function.optional_keywords.map do |name, _param|
+        Entry::OptionalKeywordParameter.new(name: name)
+      end
+    end
+
+    # TODO: add test
+    sig { params(function: RBS::Types::Function).returns(T::Array[Entry::KeywordRestParameter]) }
+    def process_rest_keywords(function)
+      param = function.rest_keywords
+
+      name = param.name || Entry::KeywordRestParameter::DEFAULT_NAME
+
+      [Entry::KeywordRestParameter.new(name: name)]
```

Comment:
I'm disliking this pattern of tiny methods more and more. Can we at least redesign this so that we are not forced to create arrays just to concatenate in the call site?